Feat: extend CSV support with settings; update test docs (#2663)
TobikoData/sqlmesh: docs/concepts/tests.md

        num_orders: 2
```

## Defining unit test data as CSV

> In unit tests, you can define data using CSV format, a common and user-friendly alternative to the more verbose YAML. Data in these supported formats can be specified inline or through external files, allowing for reuse across multiple tests.

Reviewer: ```suggestion
In unit tests, you can define data using the CSV format, a common alternative to YAML. Data in the supported formats can be specified inline or through external files, allowing for reuse across multiple tests.
```

### Inline Data

> Both inputs and outputs can be formatted as CSV. You can provide the CSV data within the YAML under the `rows:` key:

Reviewer: ```suggestion
Both inputs and outputs can be formatted as CSV. You can provide the CSV data inline under the `rows:` key:
```

```yaml linenums="1"
test_example_full_model:
  model: sqlmesh_example.full_model
  inputs:
    sqlmesh_example.incremental_model:
      format: csv
      rows: |
        id,item_id
        1,1
        2,1
        3,2
```

### External File

To specify data using an external file, use the `path` attribute:

```yaml linenums="1"
test_example_full_model:
  model: sqlmesh_example.full_model
  inputs:
    sqlmesh_example.incremental_model:
      format: csv
      path: filepath/test_data.csv
```

> In addition, data can also be loaded from an external YAML file. When the `format` is omitted, this is the default format for files.

Reviewer: ```suggestion
In addition, data can also be loaded from an external YAML file. When `format` is omitted, this is the default format for files.
```

      ...
```

When the input format is `csv`, the data can be specified inline under `rows` :

```yaml linenums="1"
    <upstream_model>:
      rows: |
        <column1_name>,<column2_name>
        <row1_value>,<row1_value>
        <row2_value>,<row2_value>
```

### `<test_name>.inputs.<upstream_model>.format`
  
The optional `format` key allows for control over how the input data is loaded. The default format is YAML, while CSV is also supported.

```yaml linenums="1"
    <upstream_model>:
      format: csv
      path: filepath/test_data.csv
```

### `<test_name>.inputs.<upstream_model>.csv_settings`
  
When the`format` is CSV, you can control the behaviour of data loading under `csv_settings`:

```yaml linenums="1"
    <upstream_model>:
      format: csv
      csv_settings: 
        sep: "#"
        skip_blank_lines: true
      rows: |
        <column1_name>#<column2_name>
        <row1_value>#<row1_value>
        <row2_value>#<row2_value>
```

> [Learn more about the supported CSV settings](https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html).

Reviewer: ```suggestion
Learn more about the [supported CSV settings](https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html).
```
